sphinxgallery/tests/test_gen_gallery.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Author: Óscar Nájera
+# License: 3-clause BSD
+"""
+Testing the rst files generator
+"""
+from __future__ import division, absolute_import, print_function
+import sphinxgallery.docs_resolv as sg
+import tempfile
+from nose.tools import assert_equals
+
+
+def _get_data(fid, shelve_file, test_string):
```

Comment:
I am not sure this helper function helps that much:
- naming isn't great, same function exists in docs_resolv and it doesn't give you a hint that this actually tests something
- when you are glancing at test_shelve it is not immediately clear where the assert actually happens

Basically I would just replace the function call by its code in test_shelve.